### PR TITLE
Added ECR Public endpoint reference

### PR DESCRIPTION
**Pull request by ShubhraDeshpande**
### What does this PR do?

Adds AWS ECR public Datadog pull urls to the documentation

### Motivation

AWS ECR Public is a public registry. Datadog is one of the popular images on the public gallery (https://gallery.ecr.aws/datadog/agent). This PR adds ECR Public to the official documentation of Datadog.

### Preview

https://github.com/ShubhraDeshpande/documentation/blob/8d9d8d48ba790ad3d9a765df1e5e998819356c0f/content/en/agent/docker/_index.md

### Additional Notes
Datadog ECR public links:
https://gallery.ecr.aws/datadog/agent
https://gallery.ecr.aws/datadog/docker-dd-agent

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/docker/_index.md [hunk 1]
```diff
@@ -50,7 +50,13 @@ If you haven’t installed the Docker Agent, follow the [in-app installation ins
 docker run -d --name dd-agent -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY=<DATADOG_API_KEY> gcr.io/datadoghq/agent:7
 ```
 
-**Note**: If you're using a different registry besides GCR, make sure to update the image.
+for ECR-public
```
Comment (sarina-dd) [inline content/en/agent/docker/_index.md:53]:
```suggestion
For ECR-public:
```

File: content/en/agent/docker/_index.md [hunk 2]
```diff
@@ -60,12 +66,22 @@ For Amazon Linux < v2:
 ```shell
 docker run -d --name dd-agent -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY=<DATADOG_API_KEY> gcr.io/datadoghq/agent:7
 ```
+For ECR-public
```
Comment (sarina-dd) [inline content/en/agent/docker/_index.md:69]:
```suggestion
For ECR-public:
```

File: content/en/agent/docker/_index.md [hunk 3]
```diff
@@ -60,12 +66,22 @@ For Amazon Linux < v2:
...
+```shell
+docker run -d --name dd-agent -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY=<DATADOG_API_KEY> public.ecr.aws/datadog/agent:7
+```
 
 For Amazon Linux v2:
 
 ```shell
 docker run -d --name dd-agent -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY=<DATADOG_API_KEY> gcr.io/datadoghq/agent:7
 ```
+For ECR-public
```
Comment (sarina-dd) [inline content/en/agent/docker/_index.md:80]:
```suggestion
For ECR-public:
```

File: content/en/agent/docker/_index.md [hunk 4]
```diff
@@ -84,6 +105,11 @@ docker run -d --name dd-agent -e DD_API_KEY=<API_KEY> -v \\.\pipe\docker_engine:
 ```shell
 docker run -d --name dd-agent -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY=<DATADOG_API_KEY> gcr.io/datadoghq/agent:7 --group-add=<DOCKER_GROUP_ID>
 ```
+for ECR-Public
```
Comment (sarina-dd) [inline content/en/agent/docker/_index.md:108]:
```suggestion
For ECR-Public:

```

**Comment by ShubhraDeshpande** [comment]
I see label is failing but is blocked on review, is there any action item on my plate?

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**